src/server/http/response/ResponseBuilder.cpp

```diff
@@ -0,0 +1,100 @@
+//
+// Created by pmerku on 11/03/2021.
+//
+
+#include <stdexcept>
+#include "server/http/response/ResponseBuilder.hpp"
+#include "utils/intToString.hpp"
+#include "utils/localTime.hpp"
+
+using namespace response;
+
+// TODO maybe wrap the map into a singleton so clang-tidy doesn't complain with static storage
+const std::map<int, std::string> ResponseBuilder::_statusMap =
+		createMap<int, std::string>
+		(100, "Continue")
+		(101, "Switching Protocols")
+		(102, "Processing")
+		(103, "Early hints")
+		(200, "OK")
+		(201, "Created")
+		(202, "Accepted")
+		(203, "Non-Authorative information")
+		(204, "No Content")
+		(205, "Reset Content")
+		(206, "Partial Content")
+		(207, "Multi-Status")
+		(208, "Already Reported")
+		(226, "IM Used")
+		(300, "Multiple Choices")
+		(301, "Moved Permanently")
+		(302, "Found")
+		(303, "See Other")
+		(304, "Not Modified")
+		(305, "Use Proxy")
+		(306, "Switch Proxy")
+		(307, "Temporary Redirect")
+		(308, "Permanent Redirect")
+		(400, "Bad Request")
+		(401, "Unauthorized")
+		(402, "Payment Required")
+		(403, "Forbidden");
+
+ResponseBuilder::ResponseBuilder(const std::string &protocol) {
+	this->_response = protocol;
+}
+
+ResponseBuilder &ResponseBuilder::setStatus(int code) {
+	std::map<int, std::string>::const_iterator it = _statusMap.find(code);
+	this->_response += " " + std::string("400"); //utils::intToString(it->first);
+	this->_response += " " + it->second;
```

Comment:
the builder should have all its parts (status code, headers, body, HTTP version) as private fields, then compile everything when you run build(). so for example, setStatus() would basically be `_status = code` with validation check to see if its a valid code.

Reply:
that would make it so the parts wouldnt have to be ran in a specific order

Reply:
I will change this yes :)